tests/activity_api_reset_test.go

```diff
@@ -0,0 +1,325 @@
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	enumspb "go.temporal.io/api/enums/v1"
+	workflowservicepb "go.temporal.io/api/workflowservice/v1"
+	sdkclient "go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/workflow"
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/common/namespace"
+	"go.temporal.io/server/common/testing/testvars"
+	"go.temporal.io/server/common/util"
+	"go.temporal.io/server/tests/testcore"
+)
+
+type ActivityApiResetClientTestSuite struct {
+	testcore.ClientFunctionalSuite
+	tv                     *testvars.TestVars
+	initialRetryInterval   time.Duration
+	scheduleToCloseTimeout time.Duration
+	startToCloseTimeout    time.Duration
+
+	activityRetryPolicy *temporal.RetryPolicy
+}
+
+func (s *ActivityApiResetClientTestSuite) SetupSuite() {
+	s.ClientFunctionalSuite.SetupSuite()
+	s.OverrideDynamicConfig(dynamicconfig.ActivityAPIsEnabled, true)
+	s.tv = testvars.New(s.T()).WithTaskQueue(s.TaskQueue()).WithNamespaceName(namespace.Name(s.Namespace()))
+}
+
+func (s *ActivityApiResetClientTestSuite) SetupTest() {
+	s.ClientFunctionalSuite.SetupTest()
+
+	s.initialRetryInterval = 1 * time.Second
+	s.scheduleToCloseTimeout = 30 * time.Minute
+	s.startToCloseTimeout = 15 * time.Minute
+
+	s.activityRetryPolicy = &temporal.RetryPolicy{
+		InitialInterval:    s.initialRetryInterval,
+		BackoffCoefficient: 1,
+	}
+}
+
+func TestActivityApiResetClientTestSuite(t *testing.T) {
+	s := new(ActivityApiResetClientTestSuite)
+	suite.Run(t, s)
+}
+
+func (s *ActivityApiResetClientTestSuite) makeWorkflowFunc(activityFunction ActivityFunctions) WorkflowFunction {
+	return func(ctx workflow.Context) (string, error) {
+
+		var ret string
+		err := workflow.ExecuteActivity(workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
+			ActivityID:             "activity-id",
+			DisableEagerExecution:  true,
+			StartToCloseTimeout:    s.startToCloseTimeout,
+			ScheduleToCloseTimeout: s.scheduleToCloseTimeout,
+			RetryPolicy:            s.activityRetryPolicy,
+		}), activityFunction).Get(ctx, &ret)
+		return "done!", err
+	}
+}
+
+func (s *ActivityApiResetClientTestSuite) TestActivityResetApi_AfterRetry() {
+	// activity reset is called after multiple attempts,
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	var activityWasReset atomic.Bool
+	activityCompleteCh := make(chan struct{})
+	var activityCompleteCount atomic.Int32
+
+	activityFunction := func() (string, error) {
+		activityCompleteCount.Add(1)
+
+		if activityWasReset.Load() == false {
+			activityErr := errors.New("bad-luck-please-retry")
+			return "", activityErr
+		}
+
+		s.WaitForChannel(ctx, activityCompleteCh)
+		return "done!", nil
+	}
+
+	workflowFn := s.makeWorkflowFunc(activityFunction)
+
+	s.Worker().RegisterWorkflow(workflowFn)
+	s.Worker().RegisterActivity(activityFunction)
+
+	wfId := testcore.RandomizeStr("wfid-" + s.T().Name())
```

Comment:
If we decided to use `tv` (and your seems to have it here too), it should be used for all things like this. And also for all other pseudo-random strings that you have in these tests.

Reply:
+1